openqa-post-similarity

```diff
@@ -0,0 +1,111 @@
+# The first line is the most important, so I focus on the first lines of the errors.
+if not os.path.exists("id_msg.json"):
+    id_msg = {}
+    for path in logs:
+        with open(path, 'r') as f:
+            text = f.read()
+            anchor = 0
+            begin = text.find("Test died", anchor)
+            while begin != -1:
+                anchor = begin + 1
+                end = text.find("\n", begin)
+                id_msg[path.split("\\")[1]] = text[begin:end]
+                begin = text.find("Test died", anchor)
+    id_msg_json = json.dumps(id_msg, sort_keys=False, indent=4, separators=(',', ':'))
+    with open("id_msg.json", "w") as f:
+        f.write(id_msg_json)
+
+# If there exists the JSON file
+else:
+    with open("id_msg.json", "r") as f:
+        id_msg = json.load(f)
+    all_dirs = os.listdir(BASE_DIR + "//openqa//testresults")
+    checked_dirs = list(id_msg.keys())
+    unchecked_dirs = [i for i in all_dirs if i not in checked_dirs]
+    for job_id in unchecked_dirs:
+        path = BASE_DIR + "//openqa//testresults//" + job_id + "//autoinst-log.txt"
+        with open(path, 'r') as f:
+            text = f.read()
+            anchor = 0
+            begin = text.find("Test died", anchor)
+            while begin != -1:
+                anchor = begin + 1
+                end = text.find("\n", begin)
+                id_msg[path.split("\\")[1]] = text[begin:end]
+                begin = text.find("Test died", anchor)
+    id_msg_json = json.dumps(id_msg, sort_keys=False, indent=4, separators=(',', ':'))
+    with open("id_msg.json", "w") as f:
+        f.write(id_msg_json)
+
+# Compute the Levenshtein result and save it.
+# Saving the result in a file may be unnecessary.
+result = {}
+f = open("LevenshteinResult.txt", "w")
+for index, (key1, value1) in enumerate(id_msg.items()):
+    calculate = {}
+    for key2, value2 in id_msg.items():
+        if key1 == key2:
+            continue
+        calculate[key2] = Levenshtein.distance(value1, value2)
+    # print(calculate)
+    calculate_sorted = sorted(calculate.items(), key=lambda x: x[1], reverse=False)
+    # print(calculate_sorted)
+    f.write("Index: " + str(index) + "\n")
+    f.write("Original error message:\n")
+    f.write("Job ID: " + key1 + "\n")
+    f.write(value1 + "\n")
+    # print(value1)
+    f.write("matched error message (top 10):\n")
+    matched_results = []
+    for i in range(10):
+        # print(id_msg[calculate_sorted[i][0]])
+        f.write("Job ID: " + calculate_sorted[i][0] + "\n")
+        f.write(id_msg[calculate_sorted[i][0]] + "\n")
+        matched_results.append(calculate_sorted[i][0])
+    f.write("\n")
+    result[key1] = matched_results
+
+# print(result)
+
+# Post the results in comments by OpenQA_Client
+client = OpenQA_Client(server="http://127.0.0.1:9526")
+for origin, matched in result.items():
+    data = {'bugrefs': [], 'created': time.strftime("%Y-%m-%d %H:%M:%S", time.localtime()) + " +0000"}
```

Comment:
It should not be necessary to pass created/updated timestamps if you just want to use the current time. Note that this code only works correctly if the local time is GMT/UTC±00:00. I assume you put this hack here to have openQA display times in your local timezone but that's not generally wanted; openQA is generally supposed to use display timestamps using GMT/UTC±00:00. So please remove that hack.